envs/env-py310.yml

```diff
@@ -7,15 +7,20 @@ dependencies:
   - bloptools >=0.7.0
   - blosc-hdf5-plugin
   - conftrak >=0.0.9
+  - cookiecutter
   - dask >=2023.9.0
   - distributed >=2023.9.0
   - databroker >=2.0.0b45
   - diffpy.pdffit2
+  - diffpy.pdfgui
   - diffpy.structure
+  - event-model>=1.21
   - hdf5-external-filter-plugins
+  - julia
   - maggma >=0.66
   - mp-api >=0.41.2
   - pychx >=4.3.1
+  - pycryptodome
```

Comment:
While I am glad `pycryptodome` is added in `env-py3{10,11,12}.yml` I have a general question about this type of packages.

It seems `pycryptodome` is already a dependency in `nslsii` (https://github.com/NSLS-II/nslsii/blob/2a31a9950d1fc266ae212c5c64ab6f83bab447b0/requirements.txt#L18) but for whatever reason the conda env is not resolving to that version of `nslsii`, so not picking up `pycryptodome`. If newer packages are installed on an overlay do we generally expect these conda environments to carry the dependencies to support them, or should the dependencies be installed on the overlay itself?

If we allow some "future" dependencies out of necessity it would be good to keep a list of them somewhere. Just a suggestion.

Alternatively, should we pin `nslsii` to a newer version instead?

Reply:
I think we never went back to https://github.com/conda-forge/nslsii-feedstock/blob/main/recipe/meta.yaml and added the requirement. Please feel free to submit a PR there.

Reply:
There seem to be few other dependencies missing in https://github.com/conda-forge/nslsii-feedstock/blob/main/recipe/meta.yaml. For example, `httpx, msgpack, msgpack-numpy, packaging, redis-json-dict` are also missing. Should they also be added to feedstock? They are all available on conda-forge.

Reply:
Let's create an issue and address it in `2024-2.4`.